### DISCUSSION TITLE
Do extensions have any mechanism to interact with cache data?

### DISCUSSION THREAD

**Discussion by abhiaagarwal** [quarto-dev/quarto-cli#9018] [Q&A]
### Description

Hi there,

I'm developing a quarto filter that pre-processes figures to optimize them (via [libvips](https://github.com/libvips/libvips))  and generate smaller variants in webp to be used with [responsive images](https://web.dev/learn/design/responsive-images) for websites. However, the act of generating an image is expensive, and I'd like to avoid re-doing it if the source image hasn't changed. Is there any built-in mechanism that will enable me to cache the result of my filter? Perhaps in the _freeze directory or something? I could hand-roll my own but I'd like to be cognizant of convention.

Thank you!

**Comment by cscheid** [comment]
None that are built-in to quarto at the moment. But Lua filters can check for the existence of files, and so it should be possible for you to check them (by attempting to open the file with `open` and checking for a `nil` value).

**Reply by abhiaagarwal** [reply]
Makes sense, thanks! Would it be considered bad practice to manually find the `_freeze` directory and manipulate it by putting the images in there? 

Otherwise, I think having a quarto idiomatic option might be a decent idea :) If you can point me to where you put the actual lua api I can make a PR with whatever I end up implementing in my own filter.

Note that [diagram](https://github.com/pandoc-ext/diagram/blob/main/_extensions/diagram/diagram.lua) in pandoc-ext uses the os-native cache directory, which I like, but isn't helpful for for avoiding costly generations in CI, for example
